lib/pbench/server/api/resources/users_api.py

```diff
@@ -8,12 +9,93 @@
 import jwt
 from sqlalchemy.exc import IntegrityError, SQLAlchemyError
 
+from pbench.server import PbenchServerConfig
 from pbench.server.api.auth import Auth
 from pbench.server.database.models.active_tokens import ActiveTokens
 from pbench.server.database.models.server_config import ServerConfig
 from pbench.server.database.models.users import User
 
 
+class AuthenticationSchema(Resource):
+    """
+    Pbench API to return authentication parameters required for the client to
+    authenticate the user with OIDC broker.
+    """
+
+    AVAILABALE_SCHEMA = ["prod", "dev", "test"]
+
+    def __init__(self, config: PbenchServerConfig, logger: Logger):
+        self.server_config = config
+        self.logger = logger
+
+    def get(self):
+        """
+        Returns the authentication schema parameters.
+
+        Client can request a specific authorization schema paramters required
```

Comment:
Yeah as I mentioned somewhere my intention was to allow the web clients to request a Keycloak broker configuration specific to a certain environment. For example, we would have a single Keycloak broker configured with multiple configurations like a staging or dev with `Pbench-dev` realm and with `Pbench-dashboard-dev` client, this configuration can be used to test out some production features we would like to deploy. We can have test env with `Pbench-test` realm and `Pbench-dashboard-test` client that can be used to do certain testing where it is okay to delete users/client or change client definition or even break the realm. And then the production environment can be completely isolated from these other environments. 

But I agree that we wouldn't technically need all these separate environments if we can just spin up a container that can import the predefined realm/client configuration and then can be destroyed later. 

> I'd originally expected that we might advertise "local", "Red Hat SS", "Google", "GitHub" schema, and provide necessary information for all of them based on the server's Keycloak configuration. (Although if all of that configuration is deferred to our captive Keycloak, the server may not even need to know about that distinction, much less advertise it to the client?)

Yeah, we will defer that to our captive Keycloak instance so we don't need these schemas in our configuration file. 

I have added a commit to revert the use of a separate endpoint for returning the authentication configuration, and it is now a part of the `endpoints` API.

Reply:
My concern is that I don't really understand how we have multiple IdM configurations that are interchangeable at client whim for a ___single___ server instance. Because otherwise there's no point in having a server know about alternatives. I don't see how that model makes any sense, but I'm happy to entertain the notion if you can explain it.

Reply:
My understanding is that we configure the Pbench Server with the URL for a single OIDC server.  This server could be a _provider_ or it could be a _broker_ for some number of providers, but the Pbench Server (nor Dashboard) doesn't care.  The Pbench Server provides the Dashboard with the URL for the OIDC thingie, the Dashboard (at the appropriate moment) loads the page at that URL, and when control returns to the Dashboard there is either a token or a failure.  It's up to the OIDC thingie to lead the user through the authentication process, perhaps loading the page for another broker or provider, and the redirects eventually bring us back to the application (i.e., to the Dashboard).

If we want multiple testing situations, then we can stand up multiple instances of the Pbench Server, and they can each be configured with the appropriate choice of OIDC server; these could point to different realms on the same Keycloak server, or they could point to different (perhaps containerized, perhaps not) Keycloak servers.  And, hopefully, they could instead point to non-Keycloak OIDC servers, but perhaps that's overly aspirational....